### PR TITLE
change stickyScrollBar onMouseMove

### PR DESCRIPTION
**Pull request by saeedjafari68**
## Summary by CodeRabbit

- **新功能**
	- 表格组件增加了更灵活的数据索引和扩展配置
	- 新增表格摘要渲染功能
	- 粘性滚动条支持不同滚动方向

- **重构**
	- 简化了表格组件的属性管理
	- 移除了部分旧的事件处理和属性

- **改进**
	- 优化了滚动条的方向处理逻辑
	- 增强了组件的配置灵活性

### CODE REVIEW

**Comment by saeedjafari68** [comment]
in [this forked repository](https://github.com/react-component/table/pull/1215) Changes have been made to correct this bug.

File: src/stickyScrollBar.tsx [hunk 1]
```diff
@@ -74,18 +75,29 @@ const StickyScrollBar: React.ForwardRefRenderFunction<unknown, StickyScrollBarPr
...
+    }else{
+      if (left > 0) {
+        left = 0;
+      }
+      if (Math.abs(left) + Math.abs(scrollBarWidth) < bodyWidth) {
+        onScroll({
+          scrollLeft: left / bodyWidth * (bodyScrollWidth + 2)
+        });
+        refState.current.x = event.pageX;
+      }
```
Comment (afc163) [inline src/stickyScrollBar.tsx:98]:
```suggestion
    const isLTR = direction === "ltr";
  
    // 限制滚动范围
    left = Math.max(isLTR ? 0 : -bodyWidth + scrollBarWidth, 
                  Math.min(isLTR ? bodyWidth - scrollBarWidth : 0, left));

    // 计算滚动位置并更新
    const shouldScroll = isLTR || Math.abs(left) + Math.abs(scrollBarWidth) < bodyWidth;
  
    if (shouldScroll) {
      onScroll({
        scrollLeft: left / bodyWidth * (bodyScrollWidth + 2)
      });
      refState.current.x = event.pageX;
    }
```

Comment (saeedjafari68) [reply src/stickyScrollBar.tsx]: Thanks
DONE.

File: src/stickyScrollBar.tsx [hunk 2]
```diff
@@ -74,19 +75,21 @@ const StickyScrollBar: React.ForwardRefRenderFunction<unknown, StickyScrollBarPr
     let left: number =
       refState.current.x + event.pageX - refState.current.x - refState.current.delta;
 
-    if (left <= 0) {
-      left = 0;
-    }
-
-    if (left + scrollBarWidth >= bodyWidth) {
-      left = bodyWidth - scrollBarWidth;
+    const isLTR = direction === "ltr";
```
Comment (afc163) [inline src/stickyScrollBar.tsx:78]:
```suggestion
    const isLTR = direction === 'ltr';
```

**Review by afc163 — APPROVED** [review]

**Comment by afc163** [comment]
@saeedjafari68 What the problem is this PR want to resolve? I can't find any issue.

**Comment by saeedjafari68** [comment]
> @saeedjafari68 What the problem is this PR want to resolve? I can't find any reproduce.

In this pull request, the sticky scrolling problem in RTL mode was solved. [In this example](https://codesandbox.io/p/sandbox/condescending-haze-j6pxcd?file=%2Findex.js), you can see the previous problem.

**Pull request merged**